overlore/db/handler.py

```diff
@@ -195,39 +188,46 @@ def init(self, path: str = "./events.db") -> DatabaseHandler:
         self.realms = Realms.instance().init()
         return self
 
-    def get_by_id(self, event_id: int) -> Any:
+    def get_by_id(self, event_id: int) -> StoredEvent:
+        # lock mutex
         self.__lock()
```

Comment:
do reads need to be thread safe? a quick google implies that theres only one writer permitted, but reads are threadsafe

Reply:
I'm not sure about reads but if it says that it's thread-safe then it should be fine. I was afraid that there would be race conditions given that we have asynchronous tasks write/reading in the db that's why I added the locks

Reply:
skimming thru this: https://ricardoanderegg.com/posts/python-sqlite-thread-safety/, i think we can simplify the locks. im fine with doing this in a follow up pr